ReduxAssignments/redux-todo-app3/src/redux/reducers/todoReducer.js

```diff
@@ -0,0 +1,39 @@
+
+import { ADD_TODO, REMOVE_TODO, UPDATE_TODO, TOGGLE_TODO_COMPLETION } from '../actions/todoActions';
+
+const initialState = {
+    todos: []
+};
+
+const todoReducer = (state = initialState, action) => {
+    switch (action.type) {
+        case ADD_TODO:
+            return {
+                ...state,
+                todos: [...state.todos, { ...action.payload, status: 'upcoming' }]
```

Comment:
deeply nested state should be shown as     state.first.second[action.someId].fourth = action.someValue